hpct-ldap-server/src/servers/ldap.py

```diff
@@ -0,0 +1,503 @@
+
+            # Private Key for Server
+            pks_args = [
+                "certtool",
+                "--generate-privkey",
+                "--bits",
+                "2048",
+                "--outfile",
+                "/etc/ldap/ldap01_slapd_key.pem",
+            ]
+            rc = subprocess.call(pks_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to create private key for server.")
+
+            # Template for Server Certificate
+            sc_template = [
+                f"organization = {org}",
+                f"cn = {result.stdout}",
+                "tls_www_server",
+                "encryption_key",
+                "signing_key",
+                "expiration_days = 365",
+            ]
+            with open("/etc/ssl/ldap01.info", "w") as f:
+                for line in sc_template:
+                    f.write(f"{line}\n")
+
+            # Create Server Certificate
+            sc_args = [
+                "certtool",
+                "--generate-certificate",
+                "--load-privkey",
+                "/etc/ldap/ldap01_slapd_key.pem",
+                "--load-ca-certificate",
+                "/etc/ssl/certs/mycacert.pem",
+                "--load-ca-privkey",
+                "/etc/ssl/private/mycakey.pem",
+                "--template",
+                "/etc/ssl/ldap01.info",
+                "--outfile",
+                "/etc/ldap/ldap01_slapd_cert.pem",
+            ]
+            rc = subprocess.call(sc_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to create server certificate.")
+
+            # Adjust Permissions and Ownership
+            g_args = ["chgrp", "openldap", "/etc/ldap/ldap01_slapd_key.pem"]
+            rc = subprocess.call(g_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to set group permissions.")
+
+            o_args = ["chmod", "0640", "/etc/ldap/ldap01_slapd_key.pem"]
+            rc = subprocess.call(o_args, stdout=subprocess.DEVNULL, stderr=subprocess.STDOUT)
+
+            if rc != 0:
+                raise Exception("Unable to set owner permissions.")
+
+            # Server Certificates LDIF
+            sc_ldif = [
+                "dn: cn=config",
+                "add: olcTLSCACertificateFile",
+                "olcTLSCACertificateFile: /etc/ssl/certs/mycacert.pem",
+                "-",
+                "add: olcTLSCertificateFile",
+                "olcTLSCertificateFile: /etc/ldap/ldap01_slapd_cert.pem",
+                "-",
+                "add: olcTLSCertificateKeyFile",
+                "olcTLSCertificateKeyFile: /etc/ldap/ldap01_slapd_key.pem",
+            ]
+            with open("/etc/ldap/certinfo.ldif", "w") as f:
+                for line in sc_ldif:
+                    f.write(f"{line}\n")
```

Comment:
What about `"\n".join(sc_ldif)`?

Reply:
I like this, can remove the for-loop without having to edit the list contents with "\n" so the original file content is clear. Updated as suggested